--- passepartout-0.2/configure.ac.orig	Thu Jun 12 17:21:38 2003
+++ passepartout-0.2/configure.ac	Tue Jul 15 12:24:05 2003
@@ -95,9 +95,9 @@
 
 # $(datadir) has to be expanded by make
 # path to install docs in
-AC_SUBST(docdir, ['"$(datadir)/doc/$(PACKAGE)/"'])
+AC_SUBST(docdir, ['"$(datadir)/doc/$(PACKAGE)-$(VERSION)/"'])
 # path to install xml stuff in
-AC_SUBST(xmldir, ['"$(datadir)/xml/$(PACKAGE)/"'])
+AC_SUBST(xmldir, ['"$(datadir)/$(PACKAGE)/xml/"'])
 #CPPFLAGS="-D'XMLPATH=\"\$(datadir)/xml/\$(PACKAGE)\"'"
 
 # Checks for libraries.
--- passepartout-0.2/src/xml2ps/Makefile.am.orig	Tue May  6 17:35:46 2003
+++ passepartout-0.2/src/xml2ps/Makefile.am	Tue Jul 15 12:35:00 2003
@@ -21,8 +21,7 @@
 xml2ps_LDADD = $(XMLPP_LIBS) $(GTKMM_LIBS) \
 	libxml2ps.a ../fonts/libfonts.a ../ps/libps.a ../util/libpptutil.a 
 
-#install DTDs and XSLTs in $(datadir)/xml/passepartout/
-xmldir = $(datadir)/xml/$(PACKAGE)
+#install DTDs and XSLTs in $(xmldir)
 dist_xml_DATA = xml2ps.dtd xhtml.xslt docbook.xslt
 
 # TODO: need updating
